lib/chef/cookbook/gem_installer.rb

```diff
@@ -45,10 +46,16 @@ def install
 
         unless cookbook_gems.empty?
           begin
-            inline_gemfile do
-              source Chef::Config[:rubygems_url]
-              cookbook_gems.each do |args|
-                gem(*args)
+            Dir.mktmpdir("chef-gem-bundle") do |dir|
+              File.open("#{dir}/Gemfile", "w+") do |tf|
+                tf.puts "source '#{Chef::Config[:rubygems_url]}'"
+                cookbook_gems.each do |args|
+                  tf.puts "gem #{args.map { |i| "'#{i}'" }.join(' ')}"
+                end
+                tf.close
+                Dir.chdir(dir) do
+                  so = shell_out!("bundle install")
```

Comment:
That sounds right. We can rely on the appbundle stub on `chef-client` itself to bust rvm envvars I think. I'm sure there is some way this can still be broken but ¯\_(ツ)_/¯